lib/galaxy/jobs/output_checker.py

```diff
@@ -146,10 +147,12 @@ def check_output(stdio_regexes, stdio_exit_codes, stdout, stderr, tool_exit_code
             if stderr:
                 state = DETECTED_JOB_STATE.GENERIC_ERROR
 
-        if DETECTED_JOB_STATE != DETECTED_JOB_STATE.OK and stderr:
+        if state != DETECTED_JOB_STATE.OK and stderr:
             if stderr:
                 peak = stderr[0:ERROR_PEAK]
-                log.debug("job failed, standard error is - [%s]" % peak)
+            else:
+                peak = ""
+            log.debug("job failed, detected %s, standard error is - [%s]" % peak)
```

Comment:
I'm not sure what you want put in `detected %s`, but you need to add one more string

Reply:
ups